### PR TITLE
Refactor neuroscope `add_recording_extractor_properties`

### PR DESCRIPTION
**Pull request by h-mayorquin**
I was working on debugging https://github.com/catalystneuro/neuroconv/pull/466 I was having a hard time with the double nested loops that we have to set the extractor properties for Neuroscope. I created a version of this function for myself for debugging the issue without the double nests and I think is more readable.

Also, @CodyCBakerPhD, do I understand by your comment on https://github.com/catalystneuro/neuroconv/pull/466#issuecomment-1575744424 that you think that the `shank_electrode_number` should be removed or do I misread?

### CODE REVIEW

**Comment by h-mayorquin** [comment]
This is a first for me. The last two commits are not showing here:
https://github.com/catalystneuro/neuroconv/compare/main...refactor_neuroscope

**Comment by CodyCBakerPhD** [comment]
> Also, @CodyCBakerPhD, do I understand by your comment on https://github.com/catalystneuro/neuroconv/pull/466#issuecomment-1575744424 that you think that the shank_electrode_number should be removed or do I misread?

That's a misread; I'm fine with the extra property `shank_electrode_number` (IDK how useful it is in general, but we've always included it for this format)

My comment there is about the values of the `channel_name` and how those values are of the form `ch{channel_id}grp{group_id}`, which I would prefer to just be `ch{channel_id}` or something similar/better looking

Basically, the `grp{group_id}` is redundant to have in the channel name since it's a parallel property for each channel

**Comment by CodyCBakerPhD** [comment]
> This is a first for me. The last two commits are in the branch but showing here. Maybe something CI related? Maybe a github problem?

Not sure I understand?

File: src/neuroconv/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py [hunk 1]
```diff
@@ -30,32 +30,31 @@ def subset_shank_channels(recording_extractor, xml_file_path: str):
...
+def add_recording_extractor_properties(recording_extractor, gain: Optional[float] = None):
     """Automatically add properties to RecordingExtractor object."""
 
     if gain:
         recording_extractor.set_channel_gains(gain)
 
-    channel_groups = get_channel_groups(xml_file_path=xml_file_path)
+    channel_ids = recording_extractor.get_channel_ids()
+    channel_names = recording_extractor.get_property(key="channel_name")
+    channel_groups = [int(name.split("grp")[1]) for name in channel_names]
```
[the remark this answers is not in the document]
Comment (h-mayorquin) [reply src/neuroconv/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py:41]: It it s producing the same in my conversion example.

Both neo and the `get_channel_groups` function we have are extracting the groups form the same format as per this comment:

https://github.com/catalystneuro/neuroconv/pull/466#issuecomment-1581220235

Everything else is just remaping from there.

**Comment by h-mayorquin** [comment]
> > This is a first for me. The last two commits are in the branch but showing here. Maybe something CI related? Maybe a github problem?
> 
> Not sure I understand?

For a while my last two commits were not dispalying here even if they were above in the branch.

**Comment by CodyCBakerPhD** [comment]
> For a while my last two commits were not dispalying here even if they were above in the branch.

But it's fixed now?

File: src/neuroconv/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (CodyCBakerPhD) [reply src/neuroconv/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py:41]: OK then

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**